lib/SimplifyPointerBitcastPass.cpp

```diff
@@ -44,9 +44,11 @@ clspv::SimplifyPointerBitcastPass::run(Module &M, ModuleAnalysisManager &) {
   return PA;
 }
 
-bool clspv::SimplifyPointerBitcastPass::runOnGEPFromBitcastCstExpr(
+bool clspv::SimplifyPointerBitcastPass::runOnInstFromCstExpr(
```

Comment:
I guess I would have expected something more generic here to just flatten out all constant expressions. Along the lines of:
```
for all i in instructions {
  for (auto o : i.operands()) {
    if (isa<ConstantExpr>(o)) {
      worklist.push_back(std::make_pair(o, i));
    }
  }
}

while !worklist.empty() {
  ConstantExpr *ce = worklist.back().first;
  Instruction *insert = worklist.back().second;
  worklist.pop_back();

  auto new_i = ce->getAsInstruction(insert);
  for (auto o : new_i->operands()) {
    if (isa<ConstantExpr>(o)) {
      worklist.push_back(std::make_pair(o, new_i));
    }
  }
}
```

Is there something I'm missing? I could also see this moving to a utility or separate pass if it's more appropriate.

Reply:
I have created a utility function in BitcastUtils that is now used in SimplifyPointerBitcast as well as LongVectorLowering and ThreeElementVectorLowering.